ui/src/app/workflows/components/events-panel.tsx

```diff
@@ -46,6 +47,44 @@ export const EventsPanel = ({namespace, name, kind}: {namespace: string; name: s
         return () => lw.stop();
     }, [showAll, hideNormal]);
 
+    const tableRef = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+        const calculateTooltips = () => {
+            const table = tableRef.current;
+
+            if (table) {
+                const columns = table.querySelectorAll<HTMLDivElement>('.argo-table-list__row .columns');
+
+                // querySelectorAll does not return an array, it returns a NodeList which is not necessarily
+                // iterable in older browsers.  As of 2/9/21, we target es5 with typescript, meaning it's
+                // not valid to use `for of` loops with NodeList.  When that target is changed or if
+                // downlevelIteration is enabled, we can swap this to use a `for of` loop.
+
+                // tslint:disable-next-line:prefer-for-of
+                for (let i = 0; i < columns.length; i++) {
+                    const col = columns[i];
+
+                    if (col.scrollWidth > col.clientWidth) {
+                        col.title = col.textContent;
+                    } else {
+                        col.title = '';
+                    }
+                }
+            }
+        };
+
+        const [debouncedCalculateTooltips, cleanup] = debounce(calculateTooltips, 1000);
+
+        window.addEventListener('resize', debouncedCalculateTooltips);
+        calculateTooltips();
```

Comment:
Seems like this is called twice? One in `debounce` and one here?

Reply:
Correct - this is immediately called in order to set up tooltips based on your current window size, then we call it in a debounced event handler on window resize to recalculate whether or not the tooltips need to be displayed.  We debounce it in order to avoid layout thrashing since we access properties like `clientWidth` and `scrollWidth` in the function.